library/core/src/task/wake.rs

```diff
@@ -303,6 +545,8 @@ impl fmt::Debug for Context<'_> {
 pub struct ContextBuilder<'a> {
     waker: &'a Waker,
     local_waker: &'a LocalWaker,
+    provider: Option<&'a mut (dyn Provider + 'static)>,
+    parent_provider: Option<&'a mut (dyn Provider + 'static)>,
```

Comment:
Shouldn't it support more levels of inheritance?

Reply:
The intention was to traverse the chain of contexts/providers but now that you mention it I realized the code is not actually doing this.